contracts/base/LenderInfo.sol

```diff
@@ -0,0 +1,216 @@
+
+    ZDAIInterface public zdai;
+
+    // array of all lending accounts
+    mapping (address => ZeroCollateralCommon.LendAccount) public lenderAccounts;
+
+    /** Modifiers */
+
+    modifier isZDai(address anAddress) {
+        require(areAddressesEqual(address(zdai), anAddress), "Address has no permissions.");
+        _;
+    }
+
+    modifier isDaiPool(address anAddress) {
+        require(areAddressesEqual(address(daiPool), anAddress), "Address has no permissions.");
+        _;
+    }
+
+    modifier isValid(address anAddress) {
+        anAddress.requireNotEmpty("Address is required.");
+        _;
+    }
+
+    /* Constructor */
+
+    constructor(
+        address zdaiAddress,
+        address daiPoolAddress
+    )
+    public
+    {
+        require(zdaiAddress != address(0x0), "ZDai address is required.");
+        require(daiPoolAddress != address(0x0), "Dai pool address is required.");
+        zdai = ZDAIInterface(zdaiAddress);
+        daiPool = DAIPoolInterface(daiPoolAddress);
+        blockAccruedInterest = block.number;
+    }
+
+    /** External Functions */
+
+    function zDaiTransfer(address sender, address recipient, uint256)
+        external
+        isZDai(msg.sender)
+        isValid(sender)
+        isValid(recipient)
+        returns (bool)
+    {
+        // Updating accrued interest for zDAI sender.
+        updateAccruedInterestFor(sender);
+        
+        // Updating accrued interest for zDAI recipient.
+        updateAccruedInterestFor(recipient);
+
+        return true;
+    }
+
+    function zDaiMinted(address recipient, uint256)
+        external
+        isDaiPool(msg.sender)
+        isValid(recipient)
+        returns (bool)
+    {
+        // Updating accrued interest for zDAI recipient.
+        updateAccruedInterestFor(recipient);
+
+        return true;
+    }
+
+    function zDaiBurnt(address recipient, uint256)
+        external
+        isDaiPool(msg.sender)
+        isValid(recipient)
+        returns (bool)
+    {
+        // Updating accrued interest for zDAI recipient.
+        updateAccruedInterestFor(recipient);
+
+        return true;
+    }
+
+    function withdrawInterest(address recipient, uint256 amount)
+        external
+        isDaiPool(msg.sender)
+        isValid(recipient)
+        returns (uint256)
+    {
+        // Updating accrued interest for zDAI recipient.
+        updateAccruedInterestFor(recipient);
+
+        uint256 amountToWithdraw = amount;
+        uint256 accruedInterest = lenderAccounts[recipient].totalAccruedInterest;
+        
+        if (accruedInterest >= amountToWithdraw) {
+            lenderAccounts[recipient].totalAccruedInterest = accruedInterest.sub(amountToWithdraw);
+        } else {
+            amountToWithdraw = lenderAccounts[recipient].totalAccruedInterest;
+            lenderAccounts[recipient].totalAccruedInterest = 0;
+        }
+        lenderAccounts[recipient].lastBlockAccrued = this.getCurrentBlockNumber();
```

Comment:
For what reason have you created `this.getCurrentBlockNumber` as a function?
It feels to me that it would be a lot simpler to simply have
```
lenderAccounts[recipient].lastBlockAccrued = block.number;
```

Reply:
It makes the code cleaner/simpler, and it allows me to mock the function (so, mocking the block.number) for the unit tests (and create different scenarios). This is one way to test contracts when they are using timestamps/block numbers/etc. Also, you can use the dependency 'mock-contract' (from Gnosis), but it depends on what you are testing.